### PR TITLE
AF: Lion optimizer on vol-10x+EMA champion (AdamW→Lion transfer)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The TandemFoil champion uses Lion optimizer with great success (val=21.319, PR #3185). The AF champion uses AdamW lr=6e-4. **Lion has never been tested on the current AF vol-10x+EMA config.** The DM dead-end for Lion was under very different conditions (no EMA, no gc, different architecture scale). AF's 2L/256d model with EMA=0.999 and gc=1.0 may be a better fit for Lion's sign-based updates.

Lion (Chen et al., 2023) uses sign(momentum) for parameter updates, producing uniform magnitude updates across all parameters. This has two potential benefits for AF volume closure:
1. **More uniform gradient signal across surface vs volume losses** — AdamW's adaptive LR can create magnitude imbalance between the 10x-weighted volume gradients and surface gradients. Lion's sign updates normalize this automatically.
2. **Lower memory footprint** — Lion tracks only momentum (1 state), not momentum + second moment (2 states), potentially allowing larger effective batch sizes or more memory for activations.

The key challenge is finding the right LR. TF uses Lion lr=1.25e-4 with gc=0.2. For AF with gc=1.0, we need to search around that neighborhood. Lion typically needs 3-10x lower LR than AdamW.

## Instructions

**Smoke test first (3 epochs)** — verify loss finite, gradients stable, no divergence.

**Trial 1 — Lion lr=1e-4 (TF-calibrated):**
```bash
cd target/icml2026 && python train.py \
  --dataset airfrans --airfrans-task full \
  --optimizer lion --lr 1e-4 \
  --cosine-t-max 50 \
  --grad-clip 1.0 \
  --weight-decay 1e-2 \
  --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 \
  --ema-decay 0.999 \
  --vol-loss-weight 10.0 \
  --wandb_group af-lion-vol10x
```

**Trial 2 — Lion lr=5e-5 (conservative):**
```bash
cd target/icml2026 && python train.py \
  --dataset airfrans --airfrans-task full \
  --optimizer lion --lr 5e-5 \
  --cosine-t-max 50 \
  --grad-clip 1.0 \
  --weight-decay 1e-2 \
  --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 \
  --ema-decay 0.999 \
  --vol-loss-weight 10.0 \
  --wandb_group af-lion-vol10x
```

**Trial 3 — Lion lr=2e-4 (aggressive, test upper bound):**
```bash
cd target/icml2026 && python train.py \
  --dataset airfrans --airfrans-task full \
  --optimizer lion --lr 2e-4 \
  --cosine-t-max 50 \
  --grad-clip 1.0 \
  --weight-decay 1e-2 \
  --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 \
  --ema-decay 0.999 \
  --vol-loss-weight 10.0 \
  --wandb_group af-lion-vol10x
```

Report `val_primary/surface_mse` AND `full_val/volume_mse` for all trials. Target: vol_mse < **0.0017** without regressing surface_mse above **0.000296**.

## Baseline

Current best AirfRANS metrics (PR #3135):

| Metric | Value | Epoch |
|--------|-------|-------|
| val_primary/surface_mse | **0.000296** | 704 |
| val_primary/vol_mse | **0.002039** | 743 |

- W&B run: `sh2zyfwr`
- Config: 2L/256d/4H, AdamW lr=6e-4, T_max=50, gc=1.0, WD=1e-2, EMA=0.999, vol-weight=10x, Fourier

Reproduce:
```bash
cd target/icml2026 && python train.py \
  --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 \
  --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 \
  --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --ema-decay 0.999 --vol-loss-weight 10.0
```

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3274 because ADVISOR: Closing per issue #3283 directive: 'Stop AirfRANS Lion.' Reassigning to priority lane..

**Comment by tcapelle** [comment]
STUDENT norman:

## Results

PR was closed by advisor per issue #3283 directive before results were ready, but all 3 trials completed. Posting for the record.

### Summary: Lion is significantly worse than AdamW on this AF config. All trials diverged.

| Trial | LR | Best Epoch | val surface_mse | val vol_mse | Diverged? |
|-------|-----|-----------|----------------|------------|-----------|
| T1 (TF-calibrated) | 1e-4 | 294 | 0.001542 | 0.01057 | Yes (NaN grads) |
| **T2 (conservative)** | **5e-5** | **376** | **0.000632** | **0.006343** | **Yes (NaN grads)** |
| T3 (aggressive) | 2e-4 | 25 | 0.02779 | 0.1063 | Yes (∞ grads) |
| **Baseline (AdamW)** | **6e-4** | **704** | **0.000296** | **0.002039** | **No** |

### Best trial (T2, lr=5e-5) vs baseline:
- **surface_mse:** 0.000632 vs 0.000296 → **2.1x worse** ❌
- **vol_mse:** 0.006343 vs 0.002039 → **3.1x worse** ❌

### Trial details

**Trial 1 — Lion lr=1e-4 (TF-calibrated):**
- W&B: `frhpn4g7`
- Best at ep294: surface_mse=0.001542, vol_mse=0.01057
- Diverged after ep294 (NaN grad norms). EMA preserved best checkpoint.
- 746 epochs trained (360-min timeout)
- Command: `CUDA_VISIBLE_DEVICES=0 python train.py --dataset airfrans --airfrans-task full --optimizer lion --lr 1e-4 --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 --epochs 999 --ema-decay 0.999 --volume-loss-weight 10.0 --agent norman --wandb-name "norman/af-lion-lr1e4" --wandb-group "af-lion-vol10x"`

**Trial 2 — Lion lr=5e-5 (conservative, BEST of 3):**
- W&B: `xb2mcog1`
- Best at ep376: surface_mse=0.000632, vol_mse=0.006343
- Diverged after ep376 (NaN grad norms). EMA preserved best checkpoint.
- 754 epochs trained (360-min timeout)
- Command: `CUDA_VISIBLE_DEVICES=1 python train.py --dataset airfrans --airfrans-task full --optimizer lion --lr 5e-5 --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 --epochs 999 --ema-decay 0.999 --volume-loss-weight 10.0 --agent norman --wandb-name "norman/af-lion-lr5e5" --wandb-group "af-lion-vol10x"`

**Trial 3 — Lion lr=2e-4 (aggressive, WORST):**
- W&B: `bkg2kdn7`
- Best at ep25: surface_mse=0.02779, vol_mse=0.1063
- Diverged extremely early (Infinity grad norms). 94x worse than baseline on surface.
- 741 epochs trained (360-min timeout)
- Command: `CUDA_VISIBLE_DEVICES=2 python train.py --dataset airfrans --airfrans-task full --optimizer lion --lr 2e-4 --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 --epochs 999 --ema-decay 0.999 --volume-loss-weight 10.0 --agent norman --wandb-name "norman/af-lion-lr2e4" --wandb-group "af-lion-vol10x"`

### What happened

Lion optimizer completely fails to match AdamW on this AirfRANS config. The hypothesis that Lion's sign-based updates would normalize surface vs volume gradients was not supported — in fact Lion made the imbalance worse. Key observations:

1. **All 3 LRs diverged.** Even the conservative lr=5e-5 diverged after ep376. The gc=1.0 clipping that stabilizes AdamW is insufficient for Lion's sign updates on this problem.
2. **Lion needs lower gc on AF.** The TF champion uses gc=0.2, while this AF config uses gc=1.0. Lion's uniform-magnitude updates amplify small gradients proportionally more than large ones, making gradient clipping less effective.
3. **Volume loss especially hurt.** Vol_mse was 3.1x worse even in the best trial — the 10x volume weighting creates gradient magnitudes that Lion's sign operator cannot handle gracefully.
4. **AdamW's adaptive LR is critical for AF.** The AF problem's multi-scale nature (surface vs volume, pressure vs velocity) benefits from per-parameter learning rate adaptation that Lion cannot provide.

### Suggested follow-ups

- Lion on AF is a dead end with the current vol-10x config. AdamW should remain the AF optimizer.
- If Lion is ever revisited for AF, gc=0.2-0.3 and lr=2-3e-5 might stabilize it, but the ceiling appears fundamentally lower than AdamW.
- The TF→AF optimizer transfer hypothesis is falsified: Lion works well on TF (1D pressure, no volume loss) but not AF (4D field, 10x volume weighting).
